DELETE, Keywords & Identifiers documents

sploiselle opened the pull request:
__Work done__
- Fleshed out `DELETE` document
- Changed 'Identifiers' document to contain information about Keywords, as well – changed name to 'Keywords & Identifiers'

__Outstanding work in this PR__
- Links from 'Identifier' now need to point to 'Keywords & Identifiers' (using appropriate header anchor link). Will do this once 'Keywords & Identifiers' is OKd

__Issues addressed__
- [Delete statement](https://github.com/cockroachdb/docs/issues/257)
- [Create document listing SQL Keywords](https://github.com/cockroachdb/docs/issues/357)
- [Explaining IDENT](https://github.com/cockroachdb/docs/issues/358)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/372)
<!-- Reviewable:end -->

jseldess commented:
Sorry for the delay here, @sploiselle. Will give this a review first thing in the morning.

sploiselle commented:
@jseldess No problem on the delay!

Sorry if you got spammed on this PR. I had a wrestling match with having removed `identifiers.md` which was updated after the PR was sent. Only after resolving that did I see `IDENT` got changed to `identifier` (🎉), so made those changes, as well.

jseldess commented:
Sean, starting with Keywords and Identifiers. Great work! Just a few comments for your to consider. 

Also, can you do a global search for links to`identifiers.html` and update that to `keywords-and-identifiers.html`?

---

Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[keywords-and-identifiers.md, line 13 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJgwNq9uivmmXckASC:-KKJgwNq9uivmmXckASD:1948800863) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L13)):*
> ```Markdown
> ## Keywords
> 
> Keywords make up SQL's vocabulary and have fixed meanings in statements. Each SQL keyword that CockroachDB supports is on one of four lists:
> ```

This is great. I like that we link to the relevant portion of the full grammar, but how about we also add a concise description of each?

---

*[keywords-and-identifiers.md, line 32 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJhtkDdTuCyUm_eXK6:-KKJhtkDdTuCyUm_eXK7:-541130865) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L32)):*
> ```Markdown
> ### Rules for Names of Objects
> 
> When naming an object, your identifier must:
> ```

This was updated a bit in https://github.com/cockroachdb/docs/pull/373. Can you factor those changes in here?

---

*[keywords-and-identifiers.md, line 40 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJin4kcXmlVv1yi9F4:-KKJin4kcXmlVv1yi9F5:-1276748982) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L40)):*
> ```Markdown
> To bypass either of the first two rules, simply surround the identifier with double-quotes. However, all references to it must also include double-quotes.
> 
> ### Rules for All Identifiers
> ```

I'm not yet clear why we need this as a separate callout. When I look at `identifier` in the grammar, in many cases, it shows up in the context of naming an object, in which case the rules for names of objects apply. Perhaps an example of `identifier` not as an object name would help? Also, if  you think we should keep this distinction, perhaps a clearer heading would be something like "Rules for Other Identifiers" instead of "All Identifiers."

@bdarnell, thoughts here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372#-:-KKJmCzIpjbTLEsLV8c8:2138590481)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Nice work on `DELETE`, Sean! Interesting in your thoughts on some of my comments.

---

Review status: 0 of 3 files reviewed at latest revision, 11 unresolved discussions.

---

*[delete.md, line 28 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJnqpY8oWKdLI4BJdi:-KKJnqpZoC65UVWq5V8p:547114090) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L28)):*
> ```Markdown
> | Parameter | Description |
> |-----------|-------------|
> |`FROM relation_expr` | The name of the table you want to delete rows from<br><br>Deleting from multiple tables in a single statement is not supported.|
> ```

Not sure we need to include the `FROM` keyword here. Also, the first sentence of the description needs a period.

---

*[delete.md, line 29 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJscU9xquKhCLn8a3K:-KKJscU9xquKhCLn8a3L:-1402574505) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L29)):*
> ```Markdown
> |-----------|-------------|
> |`FROM relation_expr` | The name of the table you want to delete rows from<br><br>Deleting from multiple tables in a single statement is not supported.|
> |`AS name` | An alias for the table name, completely hiding its original name; all subsequent references to the table must use its alias<br><br>The `AS` in this clause is optional.<br><br>Aliases are primarily used with `JOIN`, which is not yet supported but is coming in a [future release](https://github.com/cockroachdb/cockroach/issues/2970). <br>|
> ```

First sentence of the description needs a period.

---

*[delete.md, line 30 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJojF13U94I-CqL4P-:-KKJojF2MXJHpAXTTjO9:-216453839) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L30)):*
First sentence of the description needs a period. But more importantly, this is a little mislead to me. Instead of referring to the expression itself as Boolean, I like how Postgres refers to the return value as such:

> An expression that returns a value of type boolean. Only rows for which this expression returns true will be deleted.

Would you consider something more along those lines?

---

*[delete.md, line 31 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJq1d7SM2U4TCJAiuF:-KKJq1d7SM2U4TCJAiuG:1985017125) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L31)):*
> ```Markdown
> |`AS name` | An alias for the table name, completely hiding its original name; all subsequent references to the table must use its alias<br><br>The `AS` in this clause is optional.<br><br>Aliases are primarily used with `JOIN`, which is not yet supported but is coming in a [future release](https://github.com/cockroachdb/cockroach/issues/2970). <br>|
> |`WHERE a_expr`  | A Boolean expression to select the rows you want to delete<br><br/>__Without a `WHERE` clause in your statement, `DELETE` removes all rows from the table.__|
> |<code>RETURNING <em>[clauses]</em><code> | Retrieve rows deleted by the statement, using the columns identified by _`[clauses]`_<br><br>_`[clauses]`_ accepts either a `*` or `a_expr`s (1 or more)<br><br>[See an example.](#return-deleted-rows)|
> ```

I'm on the fence about how to handle these cases. Postgres lists just the parameters themselves in the parameters list, without keywords. I know I've listed keywords with parameters in such cases (`RETURNING target_list` in `INSERT`, for example). 

But the approach taken here feels overly complicated. Here are some possible options:

- One row covering both `RETURNING a_expr` and `RETURNING *`, and another row covering `AS col_label`.
- One row covering `RETURNING a_expr` with `AS col_label` covered in the description, and another row covering `RETURNING *`.

Any thoughts?

---

*[delete.md, line 100 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJtKGLdFW_YPga6gq5:-KKJtKGLdFW_YPga6gq6:1553182354) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L100)):*
> ```Markdown
> ~~~
> 
> Go ahead and `INSERT` the example values again. (Promise that we won't make you load the database a third time.)
> ```

Let's cut this. I don't think we should assume that users are reading through the examples sequentially. 

---

*[delete.md, line 119 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJtsIVgpeyhGaZAkc-:-KKJtsIVgpeyhGaZAkc0:-1358116977) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L119)):*
> ```Markdown
> ~~~
> 
> #### Return specific rows
> ```

I think this should be "Return specific columns"? 

Also, let's add the corresponding "Return all columns" above "By specifying `a`...".

---

*[delete.md, line 158 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJugucjrpJ8EinZvXf:-KKJugudi2AjnkVbavVM:-86099546) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L158)):*
> ```Markdown
> #### Delete rows using non-unique columns
> 
> Deleting rows using non-unique column values removes _all_ rows that match the expression. This can easily result in deleting data you didn't intend to.
> ```

Maybe: "that match the `WHERE` expression"?

---

*[delete.md, line 177 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJvEuPOYVThpkqU2qB:-KKJvEuPOYVThpkqU2qC:-540190464) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L177)):*
> ```Markdown
> 
> 
> ### Change returned column labels
> ```

Can we work this into the examples for `RETURNING`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372#-:-KKJvSlxLA3ScLVcsXYA:-976925649)*
<!-- Sent from Reviewable.io -->

sploiselle commented:
@jseldess Brilliant edits, Jesse. Thank you for taking the time. Will implement them and get this back to you in the next few days.

jseldess commented:
Great, @sploiselle. Thanks!

sploiselle commented:
Review status: 0 of 3 files reviewed at latest revision, 11 unresolved discussions.

---

*[delete.md, line 30 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJojF13U94I-CqL4P-:-KKQqzwrG_sMnh5RLzUP:889136433) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L30)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> First sentence of the description needs a period. But more importantly, this is a little mislead to me. Instead of referring to the expression itself as Boolean, I like how Postgres refers to the return value as such:
> 
> > An expression that returns a value of type boolean. Only rows for which this expression returns true will be deleted.
> 
> Would you consider something more along those lines?
</details>
Great advice. Because `WHERE` clauses are standard/ubiquitous, had an idea that we could try to make this as short as possible while implicitly communicating its need for Boolean values:

- Delete rows that return `TRUE` for `a_expr`

If there's some apprehension around that, could expand on it it in one of a few ways:

- Delete rows that return `TRUE` for `a_expr`. `a_expr` must be an expression that returns Boolean values
- An expression returning Boolean values; delete rows that return `TRUE`.

Does something like this work?

---

*[delete.md, line 31 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJq1d7SM2U4TCJAiuF:-KKQwIKEZ1IRpEjYU-w4:-1651813050) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L31)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> I'm on the fence about how to handle these cases. Postgres lists just the parameters themselves in the parameters list, without keywords. I know I've listed keywords with parameters in such cases (`RETURNING target_list` in `INSERT`, for example). 
> 
> But the approach taken here feels overly complicated. Here are some possible options:
> 
> - One row covering both `RETURNING a_expr` and `RETURNING *`, and another row covering `AS col_label`.
> - One row covering `RETURNING a_expr` with `AS col_label` covered in the description, and another row covering `RETURNING *`.
> 
> Any thoughts?
</details>
This was definitely too complicated. I wasn't giving the diagram enough credit for doing the heavy lifting.

Definitely like having both `RETURNING` clauses in the same line. Toying with  a few treatments for that and will have something cleaner in the next commit.

---

*[keywords-and-identifiers.md, line 13 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJgwNq9uivmmXckASC:-KKQuu-MY6K5NRii6-Ot:-545955500) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L13)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> This is great. I like that we link to the relevant portion of the full grammar, but how about we also add a concise description of each?
</details>
Descriptions for these lists are ending up a little strange because what really drives there being 4 separate lists is letting you use keywords as identifiers. I'll take a swag at it in the next commit and you let me know.

---

*[keywords-and-identifiers.md, line 40 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJin4kcXmlVv1yi9F4:-KKQxsACBH-CmHdVdZGD:-270386973) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L40)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> I'm not yet clear why we need this as a separate callout. When I look at `identifier` in the grammar, in many cases, it shows up in the context of naming an object, in which case the rules for names of objects apply. Perhaps an example of `identifier` not as an object name would help? Also, if  you think we should keep this distinction, perhaps a clearer heading would be something like "Rules for Other Identifiers" instead of "All Identifiers."
> 
> @bdarnell, thoughts here?
</details>
Had gone back and forth between "Other Identifiers" and "All Identifiers." With an example, "Other Identifiers" becomes a better option. Will include the example and changed title in the next commit.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372)*
<!-- Sent from Reviewable.io -->

bdarnell commented:
Review status: 0 of 3 files reviewed at latest revision, 11 unresolved discussions.

---

*[keywords-and-identifiers.md, line 40 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJin4kcXmlVv1yi9F4:-KKUTlLNVhxzV2MFm08T:-111338403) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L40)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary>
> Had gone back and forth between "Other Identifiers" and "All Identifiers." With an example, "Other Identifiers" becomes a better option. Will include the example and changed title in the next commit.
</details>
I don't like this division either. It places too much emphasis on the fact that we allow certain keywords to be used as identifiers in certain situations (we do this for compatibility with other sql dialects; if i were designing the language i wouldn't have so much complexity). 

I'd rather have one section for identifiers, saying something like this: 

Any string may be used as an identifier if it is surrounded by double quotes (and any double quotes it contains are doubled, i.e. `""`). The quotes may be omitted when the following conditions are true: 
- The identifier starts with a letter or underscore and contains only letters, numbers, underscores, and dollar signs
- The identifier is not equal to any relevant keyword. The *reserved* keywords are always considered relevant for the purposes of this rule; other types of keywords may or may not be relevant depending on where the keyword is used. ["relevant" isn't exactly the right word but it's the best I can come up with right now. Ideally we'd describe the rules for relevance here, but since a user can always fall back on the double-quote rule without having to understand all the details I'm fine with being vague here if we can't describe the exact rules succinctly]

Some commands may impose additional requirements on their identifiers. For example, you can't have two columns with the same name in one table. These requirements are documented on the commands' pages. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372)*
<!-- Sent from Reviewable.io -->

sploiselle commented:
Review status: 0 of 3 files reviewed at latest revision, 11 unresolved discussions.

---

*[keywords-and-identifiers.md, line 40 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJin4kcXmlVv1yi9F4:-KKV7qRWhu13wTOczPKX:-1386158068) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L40)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary>
> I don't like this division either. It places too much emphasis on the fact that we allow certain keywords to be used as identifiers in certain situations (we do this for compatibility with other sql dialects; if i were designing the language i wouldn't have so much complexity). 
> 
> I'd rather have one section for identifiers, saying something like this: 
> 
> Any string may be used as an identifier if it is surrounded by double quotes (and any double quotes it contains are doubled, i.e. `""`). The quotes may be omitted when the following conditions are true: 
> - The identifier starts with a letter or underscore and contains only letters, numbers, underscores, and dollar signs
> - The identifier is not equal to any relevant keyword. The *reserved* keywords are always considered relevant for the purposes of this rule; other types of keywords may or may not be relevant depending on where the keyword is used. ["relevant" isn't exactly the right word but it's the best I can come up with right now. Ideally we'd describe the rules for relevance here, but since a user can always fall back on the double-quote rule without having to understand all the details I'm fine with being vague here if we can't describe the exact rules succinctly]
> 
> Some commands may impose additional requirements on their identifiers. For example, you can't have two columns with the same name in one table. These requirements are documented on the commands' pages. 
</details>
It sounds like @jseldess suggests presenting naming rules as the most important piece of info; @bdarnell suggests only having identifier rules, with naming rules inferred from that. To come to an agreement on the approach the doc should take, would both of you like to hop on a call/Google Hangout for 20 minutes on Monday or Tuesday? If it's easier to hash it out internally and let me know if the direction, I'm amenable to that as well.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: 0 of 3 files reviewed at latest revision, 11 unresolved discussions.

---

*[delete.md, line 30 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJojF13U94I-CqL4P-:-KKVnRX736_q0wlJ-D0l:508787553) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L30)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary>
> Great advice. Because `WHERE` clauses are standard/ubiquitous, had an idea that we could try to make this as short as possible while implicitly communicating its need for Boolean values:
> 
> - Delete rows that return `TRUE` for `a_expr`
> 
> If there's some apprehension around that, could expand on it it in one of a few ways:
> 
> - Delete rows that return `TRUE` for `a_expr`. `a_expr` must be an expression that returns Boolean values
> - An expression returning Boolean values; delete rows that return `TRUE`.
> 
> Does something like this work?
</details>
I like the third bullet best, I think. Once we've document the where clause in more detail somewhere, we can link to that as well.

---

*[delete.md, line 31 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJq1d7SM2U4TCJAiuF:-KKVn2yR8sSDAJzPb6Ed:1434309795) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L31)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary>
> This was definitely too complicated. I wasn't giving the diagram enough credit for doing the heavy lifting.
> 
> Definitely like having both `RETURNING` clauses in the same line. Toying with  a few treatments for that and will have something cleaner in the next commit.
</details>
Great. Looking forward to it!

---

*[keywords-and-identifiers.md, line 40 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJin4kcXmlVv1yi9F4:-KKVmNMyfF86OBRVlrLc:-2146307893) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L40)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary>
> It sounds like @jseldess suggests presenting naming rules as the most important piece of info; @bdarnell suggests only having identifier rules, with naming rules inferred from that. To come to an agreement on the approach the doc should take, would both of you like to hop on a call/Google Hangout for 20 minutes on Monday or Tuesday? If it's easier to hash it out internally and let me know if the direction, I'm amenable to that as well.
</details>
@sploiselle, I really like Ben's recommendation to keep this to one section. Let's just go with that, ok? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372)*
<!-- Sent from Reviewable.io -->

sploiselle commented:
Review status: 0 of 3 files reviewed at latest revision, 11 unresolved discussions.

---

*[keywords-and-identifiers.md, line 40 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJin4kcXmlVv1yi9F4:-KKVvVgV12WvKBUM0yes:2046934075) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L40)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> @sploiselle, I really like Ben's recommendation to keep this to one section. Let's just go with that, ok? 
</details>
:thumbsup: Will do!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372)*
<!-- Sent from Reviewable.io -->

sploiselle commented:
@jseldess Integrated all of the feedback. A few things:

- `create-database.md` didn't have headers in its examples, so I added them to link the unique identifier requirement to something.
- `keywords-and-identifiers` – The descriptions of the keyword lists are a little strange. This was the shortest thing I could get to that didn't beg a ton of questions about why there were 4 lists, which ungracefully unfurls into the issue that Ben wanted to avoid–talking about naming objects using keywords.
- I considered adding an example for the keyword restriction on identifiers but decided against it to be cognizant of not pushing users in the direction of using keywords as object names.

jseldess commented:
Thanks for these improvements, Sean! We're almost good to go. Just have a few more comments/suggestions.

---

Review status: 0 of 8 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[delete.md, line 54 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKkslJb6lpawTk96d5c:-KKkslJb6lpawTk96d5d:-1078781470) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L54)):*
> ```Markdown
> DELETE FROM account_details;
> ~~~ 
> ~~~
> ```

Can you combine the request and response into a single code block? Also in the other examples?

---

*[delete.md, line 80 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKksxhWJ_2IZwavHXxw:-KKksxhWJ_2IZwavHXxx:-583625504) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L80)):*
> ```Markdown
> 
> ~~~ sql
> DELETE FROM account_details WHERE account_id=1 RETURNING *;
> ```

Totally aesthetic, but would you use `account_id = 1` instead of `account_id=1`?

---

*[delete.md, line 95 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKkt5m0f0WXozWv6X9D:-KKkt5m0f0WXozWv6X9E:-576531930) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L95)):*
> ```Markdown
> 
> ~~~ sql
> DELETE FROM account_details WHERE balance=30000 RETURNING *;
> ```

Again, pure aesthetics here, but please use `balance = 30000`.

---

*[delete.md, line 131 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKktQvHtgHw6JONffcZ:-KKktQvHtgHw6JONffc_:1178536651) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L131)):*
> ```Markdown
> 
> ~~~ sql
> DELETE FROM account_details WHERE account_id=5 RETURNING account_id, account_type;
> ```

Same as above: `account_id = 5`

---

*[delete.md, line 146 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKktVm48cG5dVFq2zQZ:-KKktVm48cG5dVFq2zQ_:-88868535) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L146)):*
> ```Markdown
> 
> ~~~ sql
> DELETE FROM account_details WHERE balance<22500 RETURNING account_id, balance AS final_balance;
> ```

Same as above: `balance < 22500`

---

*[delete.md, line 156 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKkttw6ptR3wrHMDYYs:-KKkttw7Qgq_KHLA_Nrc:861912295) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L156)):*
> ```Markdown
> ~~~
> 
> ### Create a Table Alias
> ```

Maybe rename to: "Delete Using a Table Alias"?

---

*[delete.md, line 163 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKktZ6qWLSRWVqWeRLG:-KKktZ6qWLSRWVqWeRLH:-1134326419) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L163)):*
> ```Markdown
> 
> ~~~ sql
> DELETE FROM account_details AS accounts WHERE accounts.account_id=7 RETURNING *;
> ```

Same as above: `accounts.account_id = 7`

---

*[keywords-and-identifiers.md, line 13 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJgwNq9uivmmXckASC:-KKknZV5gu62RWTHNcCF:-959837498) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L13)):*
The `Reserved Keywords` descriptions sounds a bit too extreme. Perhaps something more along the lines of: 
> Keywords that have fixed meanings and are not allowed as identifiers.

Alternately, if you prefer to not have these descriptions, we could just revert to listing the types with a follow-up paragraph along the lines of: 
> Reserved keywords have fixed meanings and are not allowed as identifiers. All other types of keywords are considered non-reserved; they have special meanings in certain contexts and can be used as identifiers in other contexts.

Your call. 

---

*[keywords-and-identifiers.md, line 40 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJin4kcXmlVv1yi9F4:-KKkqlDprntDZrZPO_SH:1658645774) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L40)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary>
> :thumbsup: Will do!
</details>
I can't quite wrap my head around the "unless the keyword is part of a list in the element's syntax" part of the second bullet. Can you help me with an example? Not sure the docs need one, but I could use the help.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372#-:-KKku6Opkdop_013dcHP:1822568831)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Reviewed 1 of 3 files at r1, 7 of 7 files at r4.
Review status: all files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372)*
<!-- Sent from Reviewable.io -->

sploiselle commented:
Two quick questions tucked away in here that once I have the answers to, I'll be able to wrap up the remaining edits.

---

Review status: all files reviewed at latest revision, 19 unresolved discussions, some commit checks failed.

---

*[create-database.md, line 23 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKi_QMLexLoYBDm0wN7:-KKi_QMLexLoYBDm0wN8:-1622590879) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/create-database.md#L23)):*
> ```Markdown
> ----------|------------
> `IF NOT EXISTS` | Create a new database only if a database of the same name does not already exist; if one does exist, do not return an error. 
> `name` | The name of the database to create, which [must be unique](#create-fails-name-already-in-use) and follow these [identifier rules](keywords-and-identifiers.html#identifiers).
> ```

Used `rename-database` as template to create headers for the examples in this doc so I could link the unique requirement to an anchor.

---

*[delete.md, line 100 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJtKGLdFW_YPga6gq5:-KKl1b3MLTHs8TDoF0Es:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L100)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Let's cut this. I don't think we should assume that users are reading through the examples sequentially. 
</details>
Done.

---

*[delete.md, line 156 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKkttw6ptR3wrHMDYYs:-KKl1ka8X2reDSmBiajn:454805719) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L156)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Maybe rename to: "Delete Using a Table Alias"?
</details>
@jseldess: Without `JOIN` aliases are funny. Technically, the alias is created, which would then ideally get used elsewhere in the statement in a non-`DELETE` context just as a reference to the table. I think the right call here might actually be to just cut the example and wait until aliases become more useful. That sound all right to you because the parameter description indicates that it's not very useful as-is?

---

*[keywords-and-identifiers.md, line 40 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJin4kcXmlVv1yi9F4:-KKl3T90b8fF9qSodYQ9:128551727) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L40)):*
Of course. The example made it a lot more clear but I feared encouraging the behavior. If you look at [`name`](https://www.cockroachlabs.com/docs/sql-grammar.html#name) (which is the primary element of the syntax where `identifier` gets used), you can see it actually lets you use keywords from 2 of the lists: `unreserved` and `col_name keywords`. If you use something on one of those lists – like `INT` – you don't need to use double-quotes and it's a valid name for the object. The converse of that is its restriction: you can't use `reserved` or `type_func_name` keywords unless you double-quote it.

Let me know if you think it's worth throwing in a clarifying example there:

> For example, `name` accepts keywords from either the Unreserved or Column Name keyword lists as identifiers.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372#-:-KKl4jAwIMEu7gDtM1hl:-2037698851)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: all files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[create-database.md, line 23 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKi_QMLexLoYBDm0wN7:-KKnQgHFDEPLoGpRmR0Q:-490100116) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/create-database.md#L23)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary>
> Used `rename-database` as template to create headers for the examples in this doc so I could link the unique requirement to an anchor.
</details>
Nice.

---

*[delete.md, line 156 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKkttw6ptR3wrHMDYYs:-KKnR5hUw1H0iNbb5Cie:1261647177) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L156)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary>
> @jseldess: Without `JOIN` aliases are funny. Technically, the alias is created, which would then ideally get used elsewhere in the statement in a non-`DELETE` context just as a reference to the table. I think the right call here might actually be to just cut the example and wait until aliases become more useful. That sound all right to you because the parameter description indicates that it's not very useful as-is?
</details>
Yep, seems best to me, too.

---

*[keywords-and-identifiers.md, line 40 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJin4kcXmlVv1yi9F4:-KKnRPuVUdsnWwUndRTa:1663740938) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L40)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary>
> Of course. The example made it a lot more clear but I feared encouraging the behavior. If you look at [`name`](https://www.cockroachlabs.com/docs/sql-grammar.html#name) (which is the primary element of the syntax where `identifier` gets used), you can see it actually lets you use keywords from 2 of the lists: `unreserved` and `col_name keywords`. If you use something on one of those lists – like `INT` – you don't need to use double-quotes and it's a valid name for the object. The converse of that is its restriction: you can't use `reserved` or `type_func_name` keywords unless you double-quote it.
> 
> Let me know if you think it's worth throwing in a clarifying example there:
> 
> > For example, `name` accepts keywords from either the Unreserved or Column Name keyword lists as identifiers.
</details>
I see. I'm just worried that the reference to "lists" in the syntax is too indirect to be immediately helpful. I think including the example would help. And maybe rephrase to something like: "Not equal any SQL keyword that is not allowed by the element's syntax. For example, ..."

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: all files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---

*[create-database.md, line 23 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKi_QMLexLoYBDm0wN7:-KKnQgHFDEPLoGpRmR0Q:-490100116) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/create-database.md#L23)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary>
> Used `rename-database` as template to create headers for the examples in this doc so I could link the unique requirement to an anchor.
</details>
Nice.

---

*[delete.md, line 156 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKkttw6ptR3wrHMDYYs:-KKnR5hUw1H0iNbb5Cie:1261647177) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L156)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary>
> @jseldess: Without `JOIN` aliases are funny. Technically, the alias is created, which would then ideally get used elsewhere in the statement in a non-`DELETE` context just as a reference to the table. I think the right call here might actually be to just cut the example and wait until aliases become more useful. That sound all right to you because the parameter description indicates that it's not very useful as-is?
</details>
Yep, seems best to me, too.

---

*[keywords-and-identifiers.md, line 40 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJin4kcXmlVv1yi9F4:-KKnRPuVUdsnWwUndRTa:1663740938) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L40)):*
<details><summary><i>Previously, sploiselle (Sean Loiselle) wrote…</i></summary>
> Of course. The example made it a lot more clear but I feared encouraging the behavior. If you look at [`name`](https://www.cockroachlabs.com/docs/sql-grammar.html#name) (which is the primary element of the syntax where `identifier` gets used), you can see it actually lets you use keywords from 2 of the lists: `unreserved` and `col_name keywords`. If you use something on one of those lists – like `INT` – you don't need to use double-quotes and it's a valid name for the object. The converse of that is its restriction: you can't use `reserved` or `type_func_name` keywords unless you double-quote it.
> 
> Let me know if you think it's worth throwing in a clarifying example there:
> 
> > For example, `name` accepts keywords from either the Unreserved or Column Name keyword lists as identifiers.
</details>
I see. I'm just worried that the reference to "lists" in the syntax is too indirect to be immediately helpful. I think including the example would help. And maybe rephrase to something like: "Not equal any SQL keyword that is not allowed by the element's syntax. For example, ..."

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372)*
<!-- Sent from Reviewable.io -->

sploiselle commented:
@jseldess Took care of last round of feedback. I'm not familiar with Reviewable, so don't know if it's worth going in and closing out all of the comments or if that'll generate a ton of noise.

Also, not sure what the expected behavior is for the CircleCI plugin. Don't know if that gripe is showing the error or if there are additional merge conflicts I need to handle.

jseldess commented:
Great. Go ahead and click "Done" or whatever for any open comments in Reviewable. There might actually be a few minor formatting things that were missed. 

Then it looks like your branch conflicts with our master, so you'll need to do `git fetch upstream` and `git merge upstream/gh-pages`, as you helped document in https://github.com/cockroachdb/docs/blob/gh-pages/CONTRIBUTING.md#keep-contributing. Since you're working in a specific local branch, I think you'd then have to bring your branch up-to-date with your local gh-pages via `git rebase gh-pages`, resolve whatever conflicts you see, and then push to the server. 

Let me know if you run into trouble. 

---

Review status: 5 of 8 files reviewed at latest revision, 19 unresolved discussions, some commit checks failed.

---

*[keywords-and-identifiers.md, line 20 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKoQpkknzFAB7WvgMBA:-KKoQpkknzFAB7WvgMBB:947231010) ([raw file](https://github.com/cockroachdb/docs/blob/9e5f4cb603671dbe75d8b3dfcb56377d949ecd45/keywords-and-identifiers.md#L20)):*
> ```Markdown
> - [Unreserved Keywords](sql-grammar.html#unreserved_keyword)
> 
> Reserved keywords have fixed meanings are not typically allowed as identifiers. All other types of keywords are considered non-reserved; they have special meanings in certain contexts and can be used as identifiers in other contexts.
> ```

"...fixed meanings are not..." should be "...fixed meanings and are not..."

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372#-:-KKoP5IXE4zhmgi1b_fg:459695371)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Disregard comment about remaining formatting issues. That's wrong. Also, don't worry about CircleCI for now.

---

Review status: 5 of 8 files reviewed at latest revision, 19 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372#-:-KKoRHcqPulXt1euRcZM:-2033894695)*
<!-- Sent from Reviewable.io -->

sploiselle commented:
Review status: 1 of 8 files reviewed at latest revision, 19 unresolved discussions, some commit checks failed.

---

*[create-database.md, line 23 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKi_QMLexLoYBDm0wN7:-KKoiPG0vKaanYik2B2Y:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/create-database.md#L23)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Nice.
</details>
Done.

---

*[delete.md, line 28 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJnqpY8oWKdLI4BJdi:-KKoiQagtDhaMoj20qgn:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L28)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Not sure we need to include the `FROM` keyword here. Also, the first sentence of the description needs a period.
</details>
Done.

---

*[delete.md, line 29 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJscU9xquKhCLn8a3K:-KKoiQzMv5A8woC2K4NZ:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L29)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> First sentence of the description needs a period.
</details>
Done.

---

*[delete.md, line 30 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJojF13U94I-CqL4P-:-KKoiRNnXaetrF2zPtpr:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L30)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> I like the third bullet best, I think. Once we've document the where clause in more detail somewhere, we can link to that as well.
</details>
Done.

---

*[delete.md, line 31 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJq1d7SM2U4TCJAiuF:-KKoiRs_t573m7dtzzQz:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L31)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Great. Looking forward to it!
</details>
Done.

---

*[delete.md, line 119 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJtsIVgpeyhGaZAkc-:-KKoiT_GuH3LWChzCaDn:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L119)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> I think this should be "Return specific columns"? 
> 
> Also, let's add the corresponding "Return all columns" above "By specifying `a`...".
</details>
Done.

---

*[delete.md, line 158 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJugucjrpJ8EinZvXf:-KKoiUUxlY72yBFDUCE1:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L158)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Maybe: "that match the `WHERE` expression"?
</details>
Done.

---

*[delete.md, line 177 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJvEuPOYVThpkqU2qB:-KKoiVJcHtuixU6QsQkq:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/delete.md#L177)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Can we work this into the examples for `RETURNING`?
</details>
Done.

---

*[delete.md, line 54 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKkslJb6lpawTk96d5c:-KKoiSX3ZveDvw0CqAEm:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L54)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Can you combine the request and response into a single code block? Also in the other examples?
</details>
Done.

---

*[delete.md, line 80 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKksxhWJ_2IZwavHXxw:-KKoiT68ORa5Kl3vY8vP:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L80)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Totally aesthetic, but would you use `account_id = 1` instead of `account_id=1`?
</details>
Done.

---

*[delete.md, line 95 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKkt5m0f0WXozWv6X9D:-KKoiTwELDNBwq5eBFdE:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L95)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Again, pure aesthetics here, but please use `balance = 30000`.
</details>
Done.

---

*[delete.md, line 131 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKktQvHtgHw6JONffcZ:-KKoiUzG3R9jxY7IkIuM:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L131)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Same as above: `account_id = 5`
</details>
Done.

---

*[delete.md, line 146 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKktVm48cG5dVFq2zQZ:-KKoiVkeuLeiobt2f10k:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L146)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Same as above: `balance < 22500`
</details>
Done.

---

*[delete.md, line 156 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKkttw6ptR3wrHMDYYs:-KKoiWVTcGdS05_lIPqb:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L156)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Yep, seems best to me, too.
</details>
Done.

---

*[delete.md, line 163 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKktZ6qWLSRWVqWeRLG:-KKoiW9bbHFp5-TsH7uG:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/c4a752d380300de0fffebc01b7bf1c91ae3299f9/delete.md#L163)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> Same as above: `accounts.account_id = 7`
</details>
Done.

---

*[keywords-and-identifiers.md, line 13 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJgwNq9uivmmXckASC:-KKoiXFo8we1Oy6X9K_8:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L13)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> The `Reserved Keywords` descriptions sounds a bit too extreme. Perhaps something more along the lines of: 
> > Keywords that have fixed meanings and are not allowed as identifiers.
> 
> Alternately, if you prefer to not have these descriptions, we could just revert to listing the types with a follow-up paragraph along the lines of: 
> > Reserved keywords have fixed meanings and are not allowed as identifiers. All other types of keywords are considered non-reserved; they have special meanings in certain contexts and can be used as identifiers in other contexts.
> 
> Your call. 
</details>
Done.

---

*[keywords-and-identifiers.md, line 32 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJhtkDdTuCyUm_eXK6:-KKoihyCBzmdPbk_KQKk:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L32)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> This was updated a bit in https://github.com/cockroachdb/docs/pull/373. Can you factor those changes in here?
</details>
Done.

---

*[keywords-and-identifiers.md, line 40 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKJin4kcXmlVv1yi9F4:-KKoiiRBcwXKTJd0ms-A:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/37bd30c590ec7e1f5d54637c73d9dfe7db481746/keywords-and-identifiers.md#L40)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> I see. I'm just worried that the reference to "lists" in the syntax is too indirect to be immediately helpful. I think including the example would help. And maybe rephrase to something like: "Not equal any SQL keyword that is not allowed by the element's syntax. For example, ..."
</details>
Done.

---

*[keywords-and-identifiers.md, line 20 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/docs/372#-KKoQpkknzFAB7WvgMBA:-KKoiYXcaNYECIWXdFku:-499146009) ([raw file](https://github.com/cockroachdb/docs/blob/9e5f4cb603671dbe75d8b3dfcb56377d949ecd45/keywords-and-identifiers.md#L20)):*
<details><summary><i>Previously, jseldess wrote…</i></summary>
> "...fixed meanings are not..." should be "...fixed meanings and are not..."
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372)*
<!-- Sent from Reviewable.io -->

sploiselle commented:
@jseldess Read something 15 times and miss an _and_. Fixed that. Resolved merge conflicts, as well.

jseldess commented:
Excellent. LGTM, Sean. Merging now.

---

Reviewed 3 of 3 files at r5, 7 of 7 files at r6.
Review status: all files reviewed at latest revision, 19 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/372#-:-KKpjWpjkMn4LaPt5P9g:1443983839)*
<!-- Sent from Reviewable.io -->
